rfcs/2021-01-05-typography-scaling.md

```diff
@@ -0,0 +1,223 @@
+
+### **Use Cases**
+There can be several use-cases when a user wants to access the dashboard on multiple devices. Few of those can be,
+- **On Mobiles** (user can check for settlement numbers while commuting/traveling in a metro/bus)
+- **On Tablets** (user can capture a payment while on-the-go to their home back from the office)
+- **On Desktops** (user can request for access to the international payment via the dashboard)
+
+Or in-general a user can visit the company's marketing/landing page and consume information about us and our offerings.
+
+### **Examples from the current system**
+Here are a few examples in the current system.
+- **Dashboard Route - Desktop vs. Mobile:**
+![Dashboard Route: Desktop vs. Mobile](./images/typography/dashboard-route-mobile-desktop-comparison.png)
+
+- **Dashboard Invoices - Desktop vs. Mobile:**
+![Dashboard Invoices: Desktop vs. Mobile](./images/typography/dashboard-invoices-mobile-desktop-comparison.png)
+
+- **Dashboard Invoices (Mobile) - Font Ratios:**
+![Dashboard Invoices (Mobile): Font Ratios](./images/typography/dashboard-invoices-mobile-problems.png)
+
+In the above examples, you can see there is no scaling across different devices/screens. Which is not considered as a good experience as we are compromising on the readability of things on different devices.
+### **Expected Outcome**
+A typography system that is more readable/consumable across different devices (mobiles/tablets/desktops).
+
+# Detailed Design
+
+### **Introduction**
+We are following a [**Major Second**](https://type-scale.com/?size=14&scale=1.125&text=A%20Visual%20Type%20Scale&font=Lato&fontweight=400&bodyfont=Poppins&bodyfontweight=400&lineheight=1.65&backgroundcolor=white&fontcolor=%23333&preview=false) modular scale (`1.125`) for our typography system on larger screens (desktops/tablets). And [**Custom**](https://type-scale.com/?size=14&scale=1.125&text=A%20Visual%20Type%20Scale&font=Lato&fontweight=400&bodyfont=Poppins&bodyfontweight=400&lineheight=1.65&backgroundcolor=white&fontcolor=%23333&preview=false) modular scale (`1.1`) for our typography system on smaller screens (mobiles). Simply put, the font-sizes will be derived in multiples of 1.125 & 1.1 for large screens and small screens respectively.
+
+### **Typefaces**
+**Base:** We will be using [**Lato**](https://fonts.google.com/specimen/Lato?sidebar.open=true&selection.family=Lato:wght@400;700) as our base font family having two weights, **Regular(400)** & **Bold(700)**.
+
+**Code:** For code modules/snippets we'll use [**Menlo**](https://en.wikipedia.org/wiki/Menlo_(typeface)) font family throughout the system having two weights, **Regular** & **Bold**.
+
+**Fallbacks:** We'll use default system fonts (according to platform) as a fallback,
```

Comment:
how did you come up with these fallbacks? I mean did you use some tool to figure out what is the closest system fallback?

Reply:
This is industry standard. I've read few blogs and put them in a particular sequence. I've already tested these on different devices.

Reply:
the problem is the users see a flash of unstyled content(a.k.a FUC) until the font is loaded. So the recommended practice is to use a font family which is close to our actual fonts to reduce this flash. You can use [this tool](https://font-match.netlify.app/)

Reply:
Yeah! I've already used this tool and system fonts are closest to Lato. Which we are also using.